language-support/hs/bindings/src/DA/Ledger/Stream.hs

```diff
@@ -93,3 +95,27 @@ isClosed Stream{status} =
     readMVar status >>= \case
         Left closed -> return (Just closed)
         Right _ -> return Nothing
+
+
+-- Here a problem with Stream is revealed: To map one stream into another requires concurrency.
+-- TODO: restructure processing to avoid the need for a sep Stream/PF
+mapListStream :: (a -> IO [b]) -> Stream a -> IO (Stream b)
```

Comment:
Yeah, my guess is something Conduit-like is the thing you need.

Reply:
Maybe. But we want concurrency. Does Conduit help here?

Reply:
Dunno, but it feels we're rolling more of our own than we should be.